src/platform/plugins/shared/discover/public/context_awareness/profile_providers/observability/traces_data_source_profile/profile.test.ts

```diff
@@ -36,34 +52,59 @@ describe('tracesDataSourceProfileProvider', () => {
 
     expect(
       tracesDataSourceProfileProvider.resolve({
+        rootContext: ROOT_CONTEXT,
         dataSource: createDataViewDataSource({ dataViewId: 'apm_static_data_view_id_custom_view' }),
         dataView: {
           getIndexPattern: () => 'traces-*',
         } as unknown as DataView,
       } as DataSourceProfileProviderParams)
     ).toEqual(RESOLUTION_MATCH);
-  });
 
-  it('should NOT match when the data source is not the APM data view', () => {
     expect(
       tracesDataSourceProfileProvider.resolve({
+        rootContext: ROOT_CONTEXT,
+        dataSource: createDataViewDataSource({ dataViewId: 'other_view_id' }),
+        dataView: { getIndexPattern: () => 'traces-*' } as unknown as DataView,
+      } as DataSourceProfileProviderParams)
+    ).toEqual(RESOLUTION_MATCH);
+
+    expect(
+      tracesDataSourceProfileProvider.resolve({
+        rootContext: ROOT_CONTEXT,
         dataSource: { type: DataSourceType.Esql },
         query: { esql: 'FROM traces' },
       } as DataSourceProfileProviderParams)
-    ).toEqual(RESOLUTION_MISMATCH);
+    ).toEqual(RESOLUTION_MATCH);
+  });
 
+  it('should NOT match when the index is not for traces', () => {
     expect(
       tracesDataSourceProfileProvider.resolve({
+        rootContext: ROOT_CONTEXT,
         dataSource: { type: DataSourceType.Esql },
         query: { esql: 'FROM logs' },
       } as DataSourceProfileProviderParams)
     ).toEqual(RESOLUTION_MISMATCH);
 
     expect(
       tracesDataSourceProfileProvider.resolve({
-        dataSource: createDataViewDataSource({ dataViewId: 'other_view_id' }),
+        rootContext: ROOT_CONTEXT,
+        dataSource: createDataViewDataSource({ dataViewId: 'other_logs_view_id' }),
         dataView: { getIndexPattern: () => 'logs-*' } as unknown as DataView,
       } as DataSourceProfileProviderParams)
     ).toEqual(RESOLUTION_MISMATCH);
   });
+
+  it("should match when the root context isn't Observability", () => {
+    expect(
+      tracesDataSourceProfileProvider.resolve({
+        rootContext: {
+          profileId: 'security-root-profile',
+          solutionType: SolutionType.Security,
+        },
+        dataSource: createDataViewDataSource({ dataViewId: 'other_view_id' }),
+        dataView: { getIndexPattern: () => 'traces-*' } as unknown as DataView,
+      } as DataSourceProfileProviderParams)
+    ).toEqual(RESOLUTION_MATCH);
+  });
```

Comment:
@elastic/security-threat-hunting-investigations (maybe @michaelolo24 specifically) I'm not sure if this was coordinated or if there would be any concerns from the Security side, but I just wanted to raise awareness that this PR will enable the traces profile outside of the O11y solution view. It's still experimental for now so users won't see it, but figured you'd want to know in advance.

Reply:
Just as reference, the PR implementation is based on [this note on the main issue for this](https://github.com/elastic/kibana/issues/208689#issuecomment-2679209210).

Reply:
This currently would activate whenever a `traces-*` index is included, or if APM is enabled, anything that also matches whatever index settings are for `spans` and `transactions`. If this is an issue, then probably this needs ironing out as to when the data source should activate.

Reply:
Yea, that would be problematic since the change would affect the view. I guess the argument becomes..what happens if there's a new mode enabled by something like `alerts-*` and both exist in the same dataView? Not to over-complicate it here, but it almost feels like we might want configuration options based on the presently selected patterns.... Can we discuss in the discover sync or a separate meeting if you didn't want to wait till then? Fyi @paulewing

Reply:
Thanks for looking @michaelolo24. I added a note to the agenda for the One Discover sync this week to discuss. I think we're probably good to proceed with this PR either way since it's experimental and not shown to users, but we'll need to figure it out before releasing in prod for sure.